### PR TITLE
fix call back handler

### PR DESCRIPTION
**Pull request by XuhuiZhou**
Closes # 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed
- [x] Branch name follows `type/descript` (e.g. `feature/add-llm-agents`)
- [x] Ready for code review

### CODE REVIEW

File: sotopia/generation_utils/generate.py [hunk 1]
```diff
@@ -482,7 +482,7 @@ async def agenerate(
         template=template,
         input_variables=input_variables,
         temperature=temperature,
-    )
+    ).with_config({"callbacks": [logging_handler]})
     if "format_instructions" not in input_values:
         input_values["format_instructions"] = output_parser.get_format_instructions()
     result = await chain.ainvoke(input_values, config={"callbacks": [logging_handler]})
```
Comment (ProKil) [inline sotopia/generation_utils/generate.py:488]: Isn't this line doing logging?

**Comment by XuhuiZhou** [comment]
Feel free to try yourself, it’s actually not. But I could be wrong

**Comment by XuhuiZhou** [comment]
So basically you want to remove the invoke config of callbacks

**Comment by ProKil** [comment]
Sorry accidentally closed, just reopened. I trust you that it was not working as expected. Just wanna dig deeper so that we are doing the right thing for the long run. 

2 questions:

1. Would this imply langchain's doc is wrong? https://python.langchain.com/v0.1/docs/modules/callbacks/ Or this was changed from v0.1 to v0.2? 
2. Can we add a test case to defend against future api changes?

**Review by ProKil — APPROVED** [review]

**Comment by XuhuiZhou** [comment]
So if you do LLMChain it’s okay to write it that way

But if you do LCEL, then it’s not working anymore :(

I am not sure why tho


Yeah, a test would be good but how? Can you give me an example?

**Comment by ProKil** [comment]
Yes,

There is an awesome fixture in pytest called `caplog`

Example: 

https://gist.github.com/hectorcanto/981b49a83a9e15b68efe5302a643e5c0
https://stackoverflow.com/questions/59742264/is-there-a-way-for-pytest-to-check-if-a-log-entry-was-made-at-error-level-or-hig

**Comment by XuhuiZhou** [comment]
@ProKil Added a test, but I don't know why it's not working

Any chance you can take a look?

**Comment by XuhuiZhou** [comment]
Okay, FYI my solution is actually wrong. I believe the `on_text` method does not work anymore
https://python.langchain.com/v0.2/docs/concepts/#callbacks

We need to modify the logger instead

**Review by ProKil — APPROVED** [review]

**Pull request merged**